frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.tsx

```diff
@@ -154,14 +158,17 @@ export const EmbedFrame = ({
     !!dashboard && isParametersWidgetContainersSticky(visibleParameters.length);
 
   const saveAsPDF = async () => {
-    const cardNodeSelector = `#${DASHBOARD_PDF_EXPORT_ROOT_ID}`;
-    await saveDashboardPdf(
-      cardNodeSelector,
-      name ?? t`Exported dashboard`,
-    ).then(() => {
-      // TODO: tracking
-      // trackExportDashboardToPDF(dashboard.id);
+    const dashboardAccessedVia = match(dashboard?.id)
+      .when(isJWT, () => "static-embed" as const)
+      .when(isUuid, () => "public-link" as const)
+      .otherwise(() => "sdk-embed" as const);
```

Comment:
Alternatively, which I prefer, is to define the return type upfront, this will allow the return type to be auto-completed as well as not having to define them as `const`.
```suggestion
      .returnType<DashboardAccessedVia>()
      .when(isJWT, () => "static-embed")
      .when(isUuid, () => "public-link")
      .otherwise(() => "sdk-embed");
```